prow/config/cmd/generate.go

```diff
@@ -44,10 +46,112 @@ func GetFileName(repo string, org string, branch string) string {
 }
 
 var (
-	inputDir  = flag.String("input-dir", "../jobs", "directory of input jobs")
-	outputDir = flag.String("output-dir", "../../cluster/jobs", "directory of output jobs")
+	inputDir        = flag.String("input-dir", "../jobs", "directory of input jobs")
+	outputDir       = flag.String("output-dir", "../../cluster/jobs", "directory of output jobs")
+	privateInputDir = flag.String("private-input-dir", "../istio-private_jobs", "directory of istio-private input jobs")
 )
 
+//updateJobSlicePrivate updates a slice of private jobs using a branch name
+func updateJobSlicePrivate(in []string, branch string) []string {
+	for key, val := range in {
+		if strings.HasSuffix(val, "_postsubmit") {
+			val = strings.Replace(val, "_postsubmit", fmt.Sprintf("_%s_postsubmit", branch), 1)
+		} else if strings.HasSuffix(val, "_presubmit") {
+			val = strings.Replace(val, "_presubmit", fmt.Sprintf("_%s_presubmit", branch), 1)
+		} else {
+			val = fmt.Sprintf("%s_%s", val, branch)
+		}
+		in[key] = val
+	}
+	return in
+}
+
+var (
+	branchModePublic  string = "public"
+	branchModePrivate string = "private"
+)
+
+//updateBranchPublic makes customizations from a main private job to a branch job
+func updateBranchPrivate(jobs genjobs.Configuration, branch string, file os.FileInfo, name string, ext string) {
+	if jobs.SupportReleaseBranching {
+		jobs.Defaults.Branches = []string{branch}
+		jobs.SupportReleaseBranching = false
+		jobs.Defaults.Modifier = strings.Replace(jobs.Defaults.Modifier, "master_", fmt.Sprintf("%s_", branch), 1)
+
+		for key, transform := range jobs.Transforms {
+			transform.JobAllowlist = updateJobSlicePrivate(transform.JobAllowlist, branch)
+			transform.JobDenylist = updateJobSlicePrivate(transform.JobDenylist, branch)
+
+			for key, val := range transform.Labels {
+				transform.Labels[key] = strings.Replace(val, "master", branch, 1)
+			}
+
+			jobs.Transforms[key] = transform
+
+		}
+
+		dst := path.Join("..", "istio-private_jobs", name)
```

Comment:
It looks you could just use the `privateInputDir` flag value instead of hardcoding it here. The tool is now generic and is used by other projects using Prow, so it's better not adding Istio specific logic.

Reply:
+1. Apologies, this is still a work in progress. Istio testing just removed the label earlier.

Reply:
I wonder if we should make the jobs version of generate.go use the `inputDir` as well. This same value is currently hard-coded in master.